### PR TITLE
 correspondence address auto look 

### PR DESCRIPTION
**Pull request by atedros-ch**
postCode look Up service added
language translation added

### CODE REVIEW

File: src/main/controllers/features/SoleTrader/soleTraderCorrespodanceAddressDetailsController.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import { FormattedValidationErrors, formatValidationError } from "../../../validation/validation";
+import { selectLang, addLangToUrl, getLocalesService, getLocaleInfo } from "../../../utils/localise";
+import { SOLE_TRADER_AUTO_LOOKUP_ADDRESS_LIST, SOLE_TRADER_AUTO_LOOKUP_ADDRESS } from "../../../types/pageURL";
+
+export const get = async (req: Request, res: Response, next: NextFunction) => {
+    req.session.user = req.session.user || {};
+    const lang = selectLang(req.query.lang);
+    const locales = getLocalesService();
+    res.render(config.SOLE_TRADER_CORRESPONDENCE_ADDRESS_LIST, {
+        title: "What is your name?",
```
Comment (swalke-ch) [inline src/main/controllers/features/SoleTrader/soleTraderCorrespodanceAddressDetailsController.ts:13]: need to correct the title

File: src/main/controllers/features/SoleTrader/soleTraderCorrespodanceAddressDetailsController.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+            res.status(400).render(config.SOLE_TRADER_CORRESPONDENCE_ADDRESS_LIST, {
+                title: "What is your correspondence address?",
+                ...getLocaleInfo(locales, lang),
+                currentUrl: SOLE_TRADER_AUTO_LOOKUP_ADDRESS_LIST,
+                previousPage: addLangToUrl(SOLE_TRADER_AUTO_LOOKUP_ADDRESS, lang),
+                pageProperties: pageProperties,
+                addresses: req.session.user.addressList
+            });
+        } else {
+            res.redirect("/sole-trader/correspondance-address-confirm");
```
Comment (swalke-ch) [inline src/main/controllers/features/SoleTrader/soleTraderCorrespodanceAddressDetailsController.ts:39]: Can we add this to pageURL.Ts

File: src/main/controllers/features/SoleTrader/soleTraderCorrespondenceAddressAutoLookupController.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        if (!errorList.isEmpty()) {
+            const pageProperties = getPageProperties(formatValidationError(errorList.array(), lang));
+            res.status(400).render(config.SOLE_TRADER_AUTO_LOOKUP_ADDRESS, {
+                title: "What is your correspondence address?",
+                ...getLocaleInfo(locales, lang),
+                currentUrl: SOLE_TRADER_AUTO_LOOKUP_ADDRESS,
+                previousPage: addLangToUrl(SOLE_TRADER_WHERE_DO_YOU_LIVE, lang),
+                pageProperties: pageProperties,
+                payload: req.body,
+                firstName: req.session.user.firstName,
```
Comment (swalke-ch) [inline src/main/controllers/features/SoleTrader/soleTraderCorrespondenceAddressAutoLookupController.ts:43]: Just checking, do we need to set name & last name in session on every page?

File: src/main/controllers/features/SoleTrader/soleTraderCorrespondenceAddressAutoLookupController.ts [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+                        formattedAddress: ukAddress.premise + ", " + ukAddress.addressLine1 + ", " + ukAddress.postTown + ", " + getCountryFromKey(ukAddress.country) + ", " + ukAddress.postcode
+                    };
+
+                    addressList.push(address);
+
+                }
+
+                req.session.user.addressList = addressList;
+                req.session.save(() => {
+                    res.redirect("/sole-trader/correspondence-address-list");
```
Comment (swalke-ch) [inline src/main/controllers/features/SoleTrader/soleTraderCorrespondenceAddressAutoLookupController.ts:86]: As I mentioned above, all URLs can move to pageURLs.ts

File: src/main/controllers/features/SoleTrader/soleTraderNameController.ts [hunk 5]
```diff
@@ -13,6 +13,7 @@ export const get = async (req: Request, res: Response, next: NextFunction) => {
 export const post = async (req: Request, res: Response, next: NextFunction) => {
     try {
         const errorList = validationResult(req);
+        console.log(errorList);
```
Comment (swalke-ch) [inline src/main/controllers/features/SoleTrader/soleTraderNameController.ts:16]: remove console.log

File: src/utils/logger.ts [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import { createLogger } from "@companieshouse/structured-logging-node";
```
Comment (swalke-ch) [inline src/utils/logger.ts:1]: there is already src/main/utils folder. Can all the file from src/utils be move there please?

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by swalke-ch — APPROVED** [review]
LGTM

**Pull request merged**